snapcraft/lifecycle.py

```diff
@@ -101,6 +101,12 @@ def _run_step(self, step, part, part_names, dirty, recursed):
                 '{}'.format(part.name, ' '.join(prereqs)))
             self.run('stage', prereqs, recursed=True)
 
+        # Prepare to run this step, if such a method is implemented
+        try:
```

Comment:
I can do that, though I feel that it makes it a bit inconsistent with how the steps are running (i.e. they check if they should run within themselves). Should I also extract that logic to here?